Core/GDCore/IDE/Events/EventsRefactorer.cpp

```diff
@@ -631,6 +632,21 @@ vector<EventsSearchResult> EventsRefactorer::SearchInEvents(
       }
     }
 
+    if (inComments) {
+      vector<gd::String> stringsVectors = events[i].GetAllSearchableStrings();
+      
+      for (std::size_t j = 0; j < stringsVectors.size(); ++j) {
+        if (!eventAddedInResults &&
+            SearchStringInComments(

```

Comment:
and there I'm not sure why you made your life much more complex that you should have ^^

You have:
* the string to search (`search`)
* the string to search in (`stringsVectors[j]`)
* the event number (`i`).

So now just search the string! :)